data-prepper-plugins/date-processor/src/main/java/com/amazon/dataprepper/plugins/processor/date/DateProcessorConfig.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.date;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+import java.time.ZoneId;
+import java.util.List;
+
+public class DateProcessorConfig {
+    static final String DEFAULT_DESTINATION = "@timestamp";
+    static final String DEFAULT_TIMEZONE = ZoneId.systemDefault().toString();
+    static final String DEFAULT_LOCALE = "en-US";
+
+    @JsonProperty("match")
+    @NotEmpty(message = "match can not be empty")
+    @Size(min = 2, message = "match should have field name and at least one pattern")
+    private List<String> match;
```

Comment:
Just a thought. It might be a better user experience if `match` is a `Map<String, List<String>>`, where the key string is the name of the event key and the `List<String>` is the format patterns to match against. I like this a little better than making the first element in the array the name of the the key. It's up to you though.

Reply:
Nice suggestion, but I think it depends on how they prefer writing configurations. I'll change it to be consistent with Grok match configuration for better user experience.

```
match: 
    timestamp: [pattern1, pattern2]
```